File: 2520-count-the-digits-that-divide-a-number/2520-count-the-digits-that-divide-a-number.cpp
class Solution {
public:
    int countDigits(int num) {
        int test = num;
        int flag;
        int counter = 0;
        while(test)
        {
            flag = test % 10;
            if(num % flag == 0)
            {
                counter++;
            }
            test /= 10;
        }
        return counter;
    }
};

File: 2520-count-the-digits-that-divide-a-number/2520-count-the-digits-that-divide-a-number.js
/**
 * @param {number} num
 * @return {number}
 */
var countDigits = function(num) {
    let counter = 0;
    let test = num;
    let temp;
    while(test)
        {
            temp = test % 10;
            if(num % temp === 0)
                {
                    counter++;
                }
            test = parseInt(test / 10);
        }
    return counter;
};